axonserver/src/main/java/io/axoniq/axonserver/message/event/EventDispatcher.java

```diff
@@ -215,15 +216,19 @@ public void appendSnapshot(String context, Authentication authentication, Event
         checkConnection(context, responseObserver).ifPresent(eventStore -> {
             try {
                 eventsCounter(context, snapshotCounter, BaseMetricName.AXON_SNAPSHOTS).mark();
-                eventStore.appendSnapshot(context, authentication, snapshot).whenComplete((c, t) -> {
-                    if (t != null) {
-                        logger.warn(ERROR_ON_CONNECTION_FROM_EVENT_STORE, "appendSnapshot", t.getMessage());
-                        responseObserver.onError(t);
-                    } else {
-                        responseObserver.onNext(c);
-                        responseObserver.onCompleted();
-                    }
-                });
+                eventStore.appendSnapshot(context, snapshot, authentication)
```

Comment:
Why put the actions in the doOnEach instead of in the subscribe()?

Reply:
Sure, I can do that, to use `subscribe(onNext, onError, onComplete)`, but there is no `onNext` handler in our case, only `onError` and `onComplete`. Felt cleaner this way...